File: xabber/src/main/java/com/xabber/xmpp/smack/XTokenRequestIQ.java
package com.xabber.xmpp.smack;

import org.jivesoftware.smack.packet.IQ;

public class XTokenRequestIQ extends IQ {

    public static final String ELEMENT   = "issue";
    public static final String NAMESPACE = "http://xabber.com/protocol/auth-tokens";

    private final String client;
    private final String device;

    public XTokenRequestIQ(String client, String device) {
        super(ELEMENT, NAMESPACE);
        this.client = client;
        this.device = device;
    }

    @Override
    protected IQChildElementXmlStringBuilder getIQChildElementBuilder(IQChildElementXmlStringBuilder xml) {
        xml.rightAngleBracket();
        xml.optElement("client", client);
        xml.optElement("device", device);
        return xml;
    }

    public String getClient() {
        return client;
    }

    public String getDevice() {
        return device;
    }
}

File: xabber/src/main/java/com/xabber/xmpp/xtoken/XTokenRequestIQ.kt
package com.xabber.xmpp.xtoken

import com.xabber.android.data.extension.xtoken.XTokenManager
import org.jivesoftware.smack.packet.IQ
import org.jxmpp.jid.DomainBareJid

class XTokenRequestIQ(
    server: DomainBareJid,
    private val client: String,
    private val device: String,
) : IQ(ELEMENT, NAMESPACE) {

    init {
        type = Type.set
        to = server
    }

    override fun getIQChildElementBuilder(xml: IQChildElementXmlStringBuilder) = xml.apply {
        rightAngleBracket()
        optElement("client", client)
        optElement("device", device)
    }

    companion object {
        const val ELEMENT = "issue"
        const val NAMESPACE = XTokenManager.NAMESPACE
    }

}